Rc v1.0.0

Ramin-Farhadi opened the pull request:
- Custom Post Type Addition:
A custom post type named "Vertical Sliders" has been added to the project.

- ACF Pro Plugin Configuration:
When using the ACF Pro plugin, ensure to add the repeater fields with the field name "vertical_sliders_fields." Define four fields within the repeater as Images, Title, Subtitle, and Description.

- Shortcode Usage Requirement:
When implementing the [ucf_vertical_slide] shortcode, include the ID number of each custom post. For example, it should be defined as [ucf_vertical_slide id="123456"].

Commit:
ucf-vertical-slide.php was added

jmbarne3 committed:
Built readme and generated package-lock

jmbarne3 committed:
Added css and js that Ramin wrote and integrated it into the gulp process

Commit:
custom post type was created

Commit:
ACF fields were created.

Commit:
echo my_json_data muted as a comment

jmbarne3 commented on ucf-vertical-slide.php:
```diff
@@ -11,3 +11,68 @@
+}
+
+// Register the shortcode
+add_shortcode( 'ucf_vertical_slide', 'ucf_vertical_slide_shortcode' );
```
Move this into a separate file under `includes/` and import it into here. This is just to keep things clean.

Ramin-Farhadi replied:
It's now in the -> includes/ucf-slide-shortcode.php

jmbarne3 commented on ucf-vertical-slide.php:
```diff
@@ -11,3 +11,68 @@
+add_shortcode( 'ucf_vertical_slide', 'ucf_vertical_slide_shortcode' );
+
+
+function ucf_vertical_slide_enqueue_assets() {
+	global $my_json_data; // Access the global variable
+	$plugin_data = get_plugin_data( UCF_VERTICAL_SLIDE__FILEPATH );
+	$version = $plugin_data['Version'];
+
+	// Enqueue styles
+    wp_enqueue_style( 'ucf-vertical-slide-styles', UCF_VERTICAL_SLIDE__STATIC_URL . '/css/style.min.css', null, $version );
+
+    // Enqueue scripts
+    wp_enqueue_script( 'ucf-vertical-slide-script', UCF_VERTICAL_SLIDE__STATIC_URL . '/js/script.min.js', array( 'jquery' ), $version, true );
+
+}
+
+add_action( 'wp_enqueue_scripts', 'ucf_vertical_slide_enqueue_assets', 10, 0 );
```
Move this into a separate file under `includes/` for clarity.

Ramin-Farhadi replied:
It's now in the -> includes/ucf-slide-assets.php

jmbarne3 commented on src/scss/_vertical-slide.scss:
```diff
@@ -0,0 +1,64 @@
+  }
+}
+
+.sliderParentRow {
```
All CSS class names need to be update to use lower case and dashes. No camel case, please.

Ramin-Farhadi replied:
Css was modified.

jmbarne3 commented on ucf-vertical-slide.php:
```diff
@@ -11,3 +11,68 @@
+	function ucf_vertical_slide_shortcode( $atts ) {
+		ob_start();
+		$defaults = array(
+			'slug' => 'default_slug',
+			'id' => 'default_id'
```
It's very likely we do not want default values on these. If the user adds a `vertical_side` shortcode without a slug or id, what behavior do we want? We probably want to return out an empty string or else throw an error (depending on if we want it to fail gracefully or loudly)/

Ramin-Farhadi replied:
Just removed the default value. Picked the gracefully method :D

jmbarne3 commented on ucf-vertical-slide.php:
```diff
@@ -11,3 +11,68 @@
+
+		// Getting the Id and Slug from shortcode
+		$atts = shortcode_atts($defaults, $atts, 'ucf_vertical_slide');
+		$slider_slug = $atts['slug'];
+		$slider_id = $atts['id'];
```
You will need to handle both of these parameters being passed in. So, if a slug is passed in, but not an ID, you need to go find the post using the slug and _get_ the id. If the id is passed in but not the slug, you're all good to go. If both are passed in, you probably want to ignore the slug and use the id.

Ramin-Farhadi replied:
I will handle this in different PR.

jmbarne3 commented on ucf-vertical-slide.php:
```diff
@@ -11,3 +11,68 @@
+
+				$sliders_objects = get_field( 'vertical_sliders_fields' , $slider_id );
+				$my_json_data = json_encode($sliders_objects);
+				echo $my_json_data;
```
This looks like debugging code that can probably be removed, since you will be getting this data from elsewhere in the future.

Ramin-Farhadi replied:
it was removed.

Commit:
php ucf-vertical-php.slide from root folder was modifed

Now with the benefit of using include_onc, the main file was seprated into three files, ucf-vertical-slide-shortcode, ucf-vertical-slide-assets and ucf-vertical-slider-posttype

Commit:
debugging was muted
